arango/executor.py

```diff
@@ -275,19 +279,34 @@ def __init__(
         if max_size is not None:
             data["maxTransactionSize"] = max_size
 
-        request = Request(
-            method="post",
-            endpoint="/_api/transaction/begin",
-            data=data,
-            headers={"x-arango-allow-dirty-read": "true"} if allow_dirty_read else None,
-        )
-        resp = self._conn.send_request(request)
+        self._id: str = ""
```

Comment:
Nitpicking: there is no need to "declare" variables in Python. This line can be safely removed.

Reply:
I agree, but `mypy` disagrees. This triggers a `mypy: no-redef` error, because `self._id` is declared with a type hint in both branches of the conditional. There is an issue for this here: https://github.com/python/mypy/issues/14534 but it appears to be something they don't want to "fix".